lib/typed/result.rb

```diff
@@ -24,5 +24,21 @@ def payload; end
 
     sig { abstract.returns(Error) }
     def error; end
+
+    sig do
+      abstract
+        .type_parameters(:U, :T)
+        .params(_block: T.proc.params(arg0: Payload).returns(Result[T.type_parameter(:U), T.type_parameter(:T)]))
+        .returns(T.any(Result[T.type_parameter(:U), T.type_parameter(:T)], Result[T.type_parameter(:U), Error]))
+    end
+    def map(&_block); end
```

Comment:
I'd like to have the functional naming in here as aliases, but advertise a name more like `and_then` which is a little easier to take in for normal Ruby users.

Reply:
Agree, I've now renamed `map` into `and_then`, we can add aliases later on if needed 👍 
Also, I've removed `flat_map` because it doesn't really bring any tangible benefit